src/hooks/useGetData.jsx

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from 'react'
+
+let endpoint = null
+
+export const useGetData = (API_BASE_URL, path, page, limit, characterId = null) => {
+   const [data, setData] = useState(null)
+   const [isLoading, setIsLoading] = useState(true)
+   const [error, setError] = useState(null)
+
+   useEffect(() => {
+      const fetchData = async () => {
+         setIsLoading(true)
+         setData(null)
+         try {
+            if (characterId) {
+               endpoint = new URL(`${path}/${characterId}`, API_BASE_URL).href
+            } else {
+               endpoint = new URL(path, API_BASE_URL).href + `?page=${page}&pageSize=${limit}`
```

Comment:
Concatenation is unnecessary here, the entire expression can be included in a template string.